src/engine/source/kvdb/src/kvdb.cpp

```diff
@@ -1,561 +1,529 @@
-                                result =  false;
-                            }
-                        }
-                        s = txn->Commit();
-                        if (s.ok()) {
-                            LOG(ERROR) << "[" << __func__ << "]" << " transaction commited OK" << std::endl;
-                            delete txn;
-                            result =  true;
-                        }
-                        else {
-                            LOG(ERROR) << "[" << __func__ << "]" << " couldn't commit transaction, error: " << s.code() << std::endl;
-                            result =  false;
-                        }
-                    }
+    if (m_state == State::Open)
+    {
+        rocksdb::Transaction *txn = m_txndb->BeginTransaction(kOptions.write);
+        if (txn)
+        {
+            for (auto pair : pairsVector)
+            {
+                std::string const key = pair.first;
+                std::string const value = pair.second;
+                if (key.empty())
+                {
+                    WAZUH_LOG_ERROR("can't write to a Transaction to a family "
+                                    "column with no key.");
+                    return false;
+                }
+                // Write a key in this transaction
+                rocksdb::Status s = txn->Put(cfh->second, key, value);
+                if (s.ok())
+                {
+                    continue;
                 }
-                handlers.clear();
+                else
+                {
+                    WAZUH_LOG_ERROR("couldn't execute Put in transaction "
+                                    "-breaking loop-, error: [{}]",
+                                    s.ToString());
+                    return false;
+                }
+            }
+            rocksdb::Status s = txn->Commit();
+            if (s.ok())
+            {
+                WAZUH_LOG_INFO("transaction commited OK");
+                delete txn;
+                return true;
             }
-            else {
-                LOG(ERROR) << "[" << __func__ << "]" << " couldn't begin transaction, error: " << s.code() << std::endl;
-                result =  false;
+            else
+            {
+                WAZUH_LOG_ERROR("couldn't commit in transaction, error: [{}]",
+                                s.ToString());
+                return false;
             }
         }
-        else {
-            LOG(ERROR) << "[" << __func__ << "]" << " couldn't Open DB, error: " << s.code() << std::endl;
-            result = false;
+        else
+        {
+            WAZUH_LOG_ERROR("couldn't begin in transaction");
+            return false;
         }
-        delete txn_db;
     }
-    else {
-        LOG(ERROR) << "[" << __func__ << "]" << " can't delete a FC that doesn't exist" << std::endl;
-        result = false;
+    else
+    {
+        WAZUH_LOG_ERROR("DB should be open for execution");
+        return false;
```

Comment:
i find this function a bit hard to follow. Consider refactoring like this (this is the way to avoid the arrow pattern mentioned above)
```
bool KVDB::writeToTransaction(
    const std::vector<std::pair<std::string, std::string>> pairsVector,
    const std::string &columnName)
{
    if (m_state != State::Open)
    {
        WAZUH_LOG_ERROR("DB should be open for execution");
        return false;
    }

    if (!pairsVector.size())
    {
        WAZUH_LOG_ERROR(
            "couldn't write transaction to DB, nedd at least 1 element");
        return false;
    }

    CFHMap::const_iterator cfh = m_CFHandlesMap.find(columnName);
    if (cfh == m_CFHandlesMap.end())
    {
        WAZUH_LOG_ERROR("couldn't write transaction to DB unknown column name");
        return false;
    }

    auto txn = std::unique_ptr<rocksdb::Transaction>(m_txndb->BeginTransaction(kOptions.write));
    if (!txn)
    {
        WAZUH_LOG_ERROR("couldn't begin in transaction");
        return false;
    }

    for (auto [key, value] : pairsVector)
    {
        if (key.empty())
        {
            WAZUH_LOG_ERROR("can't write to a Transaction to a family "
                            "column with no key.");
            return false;
        }
        // Write a key in this transaction
        rocksdb::Status s = txn->Put(cfh->second, key, value);
        if (!s.ok())
        {
            WAZUH_LOG_ERROR("couldn't execute Put in transaction "
                            "-breaking loop-, error: [{}]",
                            s.ToString());
            return false;
        }
    }

    rocksdb::Status s = txn->Commit();
    if (!s.ok())
    {
        WAZUH_LOG_ERROR("couldn't commit in transaction, error: [{}]",
                        s.ToString());
        return false;
    }

    WAZUH_LOG_DEBUG("transaction commited OK");
    return true;
}
```

Reply:
awesome!